app/src/main/java/de/westnordost/streetcomplete/data/quest/VisibleQuestsSource.kt

```diff
@@ -69,9 +71,16 @@ import javax.inject.Singleton
         }
     }
 
+    private val teamModeChangeListener = object : TeamModeQuestFilter.TeamModeChangeListener {
+        override fun onTeamModeChanged(enabled: Boolean) {
+            // TODO: Not yet implemented
```

Comment:
Here we have access to whether team mode is enabled or not, but how can we loop through all quests and update their visibility then?

Reply:
Right, this class doesn't know that. So that is why when the switch is flipped, the class should get all the visible quests from the two controllers by itself, sort them into invisible and visible quests and call the `onUpdatedVisibleQuests` listener with that info.
Yes, this may lead to already visible quests being added twice and already not-visible quests to be deleted from view twice. You need to check if the listener(s), mainly it should be the QuestPinLayerManager, are OK with that.

Hmm, but then of course,... the question is fro which bbox the visible quests should be retrieved from DB. Hmm.

I have had the same problem on my development branch, this is why I extended the listener like this:
```kotlin
    interface Listener {
        /** Called when given quests in the given group have been added/removed */
        fun onUpdatedVisibleQuests(added: Collection<Quest>, removed: Collection<Long>, group: QuestGroup)
        /** Called when something has changed which should trigger any listeners to update all */
        fun onInvalidated()
    }
```
... with an onInvalidated() method. The `QuestPinLayerManager` does this then:

```kotlin
    override fun onInvalidated() {
        clear()
        onNewScreenPosition()
    }
```

to re-fetch the new quests.